src/main/java/no/sikt/nva/BrageProcessor.java

```diff
@@ -118,14 +136,30 @@ private Optional<Record> processBundle(LicenseScraper licenseScraper,
             record.setBrageLocation(String.valueOf(brageLocation.getBrageBundlePath()));
             var warnings = BrageProcessorValidator.getBrageProcessorWarnings(entryDirectory);
             record.getWarnings().addAll(warnings);
-            logWarningsIfNotEmpty(brageLocation,warnings);
+            checkForEmbargo(record);
+            logWarningsIfNotEmpty(brageLocation, warnings);
             return Optional.of(record);
         } catch (Exception e) {
             logger.error(e.getMessage() + StringUtils.SPACE + brageLocation.getOriginInformation());
             return Optional.empty();
         }
     }
 
+    private void checkForEmbargo(Record record) {
+        var handle = record.getId().toString();
+        if (EmbargoScraper.containsHandle(embargoes, handle)) {
+            var potentialEmbargo = embargoes.stream()
+                                       .filter(embargo -> embargo.getHandle().equals(handle))
+                                       .findAny().orElse(null);
+            if (recordContainsEmbargoFile(record, Objects.requireNonNull(potentialEmbargo))) {
+                record.getContentBundle()
+                    .getContentFileByFilename(potentialEmbargo.getFilename())
+                    .setEmbargoDate(potentialEmbargo.getDate());
+                logger.info(potentialEmbargo.getDate());
+            }
+        }
+    }
+
```

Comment:
Nå er embargo listene ganske små (for Kristiania er det vel 3 poster med embargo). Men hvis man sliter med optimalisering på denne. Så ville å konvertere dette til et hashmap hvor key'en er et set av handle + filnavn, og value er embargo date.